### ISSUE TITLE
Continued support for RHEL v7

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10343]
Some OSMC members need RHEL v7 support for critical applications. We need to reinstate that ASAP, preferably before we release 1.21.0.

**Comment by sjoelund** [comment]
The weirdest parts are the devtoolset packages, which sometimes have been removed upstream and older OM releases then not being possible to install. I suppose I can simply update it to devtoolset-11 and it should be good until RHEL7 support is gone upstream.

The cmake and gcc versions are probably OK according to @mahge 

But it is worth noting that a lot more things can break for RHEL since they sometimes update packages in the middle of a cycle (and requiring devtoolset things to be installed). Debian/Ubuntu releases are not supported for as long, but installation has tended to always work even on older packages compiled for the same OS release.

**Comment by sjoelund** [comment]
Oh yeah, and we haven't removed any support yet.

**Comment by mahge** [comment]
> The [cmake and gcc versions](https://openmodelica.org/developersresources/devdocumentation/) are probably OK according to @mahge

Yes. We are all good. We can officially update our minimum requirements for building OpenModelica from source (at least for CMake) without leaving anything supported by OpenModelica behind. 

The issue right now is that we are using tools with versions much lower than what we have [here](https://openmodelica.org/developersresources/devdocumentation/) to check things. For example, the PR build check still uses `gcc-5` and `CMake-3.5` (IIRC), even though the minimum versions we need to support are `gcc-8.3.1` and `CMake-3.16.3` respectively. This, right now, can sometimes make it impossible to get anything that is not supported by those versions into OpenModelica source code because it will fail on the PR.  

We should, of course, wait until after the upcoming release to actually upgrade the images we use for checking PRs (and building nightlies and releases).

**Comment by casella** [comment]
> Oh yeah, and we haven't removed any support yet.

OK, I understood that the el7 job was removed from LINUX_BUILD, but I was mistaken, sorry. So, we are still getting a build for el7 and we'll do do until 1.21.0 is released.

> We should, of course, wait until after the upcoming release to actually upgrade the images we use for checking PRs (and building nightlies and releases).

But then, what happens to the el7 build?

**Comment by mahge** [comment]
> OK, I understood that the el7 job was removed from [LINUX_BUILD](https://test.openmodelica.org/jenkins/blue/organizations/jenkins/LINUX_BUILDS/detail/LINUX_BUILDS/2090/pipeline), but I was mistaken, sorry. So, we are still getting a build for el7 and we'll do do until 1.21.0 is released.

For my needs at least, there is no issue with RHEL v7. It already provides better versions of the tools than we currently require as minimum so we can definitely keep it around. We do not have to do anything about RHEL v7. What we need to do is update the docker images we use for testing PRs (I think some of them are based on Ubuntu Xenial (16.04) or Bionic (18.04). However, even that by itself is not a problem as long as we upgrade the tools themselves.) These images are the ones that are using old versions of the tools we need. Since they are the ones that we use for testing PRs, any change we make, that does not build on those images and their versions of the tools, can never make it into the OpenModelica repository. 

Once we make the release, we can go ahead and start updating these default PR-checking images so we can start upgrading things. RHEL v7 can continue to be supported until EOL.

**Comment by sjoelund** [comment]
The main problem I have with el7 is that it is the last rpm OS still using yum instead of dnf. (It used to have a version of dnf but it was removed for being insecure instead of bugfixed...)

**Comment by sjoelund** [comment]
> What we need to do is update the docker images we use for testing PRs (I think some of them are based on Ubuntu Xenial (16.04) or Bionic (18.04)

Yes, we should update them every 2 years. Bionic is end of life in a month and I marked it obsolete early (because why build the next OM release for it when the repositories will be gone very soon).

Hopefully #10353 works... I don't really have time to work on it.

**Comment by mahge** [comment]
> Hopefully https://github.com/OpenModelica/OpenModelica/pull/10353 works... I don't really have time to work on it.

Thank you. No worries I can finish it up.

**Comment by sjoelund** [comment]
Can this be closed? RHEL7 went EOL in June 2024
